library/core/src/lib.rs

```diff
@@ -357,4 +361,23 @@ pub mod arch {
     }
 }
 
+// Pull in the `core_simd` crate directly into libcore. The contents of
+// `core_simd` are in a different repository: rust-lang/portable-simd.
+//
+// `core_simd` depends on libcore, but the contents of this module are
+// set up in such a way that directly pulling it here works such that the
+// crate uses this crate as its libcore.
+#[path = "../../portable-simd/crates/core_simd/src/mod.rs"]
+#[allow(missing_debug_implementations, dead_code, unsafe_op_in_unsafe_fn)]
+#[allow(rustdoc::bare_urls)]
+#[unstable(feature = "portable_simd", issue = "86656")]
+mod core_simd;
+
+#[doc = include_str!("../../portable-simd/crates/core_simd/src/core_simd_docs.md")]
```

Comment:
This suggests to me that a better approach -- if we go down the subtree route -- may be to have the rust-lang/rust subtree *not* be the root of the portable-simd repository, but rather the core_simd/src directory (or something along those lines).

But I'm not sure subtrees can support that kind of subdirectory workflow...

Reply:
I don't think git subtree supports that.

Reply:
Well... that depends on how things go over time, I think. The main other crate in our workspace is just designed to assist our test suite, and while it is stochastic it is rarely super-unpredictable and not that heavyweight. I think we expected early on that we might have a number of crates, like stdarch does, but it just kinda never panned out. Maybe that will change anyways sometime soon? But it seems equally likely that we might simplify down to a single crate in the root anyways.

And when I mentioned the tests to @jyn514, he suggested infra might actually be interested in moving them out into the main repo as well, as it would have forced issues like #88769 to get resolved faster, that's for sure. I don't know if that makes sense at the moment, though, and I didn't want to make a big fuss, but it's something I am thinking about in terms of future directions.